src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -255,6 +234,38 @@ def open_result_file(filename: Path) -> h5py.File:
             f.close()
 
 
+def _open_result_file(filename: Path, h5py_file: h5py.File) -> h5py.File:
```

Comment:
- [ ] This `h5py_file` parameter does not make sense, as it will always be the same object (either `h5py.File` or a `partial` only depending on the current `h5py` version), remove the parameter and inline that code inside this function.